usr/lib/ep11_stdll/ep11_specific.c

```diff
@@ -2870,6 +2880,43 @@ static int get_curve_type_from_template(TEMPLATE *tmpl)
     return curve_type;
 }
 
+CK_BBOOL ep11_check_pkey_attrs(CK_ATTRIBUTE_PTR attrs,
+                               CK_ULONG attrs_len, CK_ULONG type)
+{
+    CK_BBOOL val;
+    CK_RV rc = CKR_OK;
+
+    rc = get_bool_attribute_by_type(attrs, attrs_len, type, &val);
+    if (rc != CKR_OK) {
+        /*  Attributes may not be specified in the attrs,
+         *  returning default TRUE for CKA_EXTRACTABLE
+         *  and FALSE for all other attributes  */
+        return type == CKA_EXTRACTABLE ? TRUE: FALSE;
+    }
+    return val;
+}
+
+CK_RV ep11tok_aes_xts_pkey_check(STDLL_TokData_t *tokdata, CK_ATTRIBUTE_PTR attrs,
+                                 CK_ULONG attrs_len)
+{
+    ep11_private_data_t *ep11_data = tokdata->private_data;
+
+    if (ep11tok_pkey_option_disabled(tokdata) == TRUE ||
+        ep11_data->msa_level < 4 ||
+        ep11tok_is_mechanism_supported(tokdata, CKM_IBM_CPACF_WRAP) != CKR_OK ||
+        ep11tok_is_mechanism_supported(tokdata, CKM_AES_KEY_GEN) != CKR_OK) {
```

Comment:
Instead of checking these 4 preconditions here, you could simply use
`if (ep11tok_is_mechanism_supported(tokdata, CKM_AES_XTS_KEY_GEN) == CKR_OK) ....` which already checks the same things (you introduces this only a few commits later, but that should not matter).